src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1029,6 +1040,23 @@ bool isOrContainsErrorType(TypeSymbol typeSymbol)
             {
                 return (object)typeSymbol.VisitType((currentTypeSymbol, unused1, unused2) => currentTypeSymbol.IsErrorType(), (object)null) != null;
             }
+
+            static bool isValidNullableOverride(
+                CSharpCompilation compilation,
+                RefKind refKind,
+                TypeWithAnnotations sourceType,
+                TypeWithAnnotations targetType)
+            {
+                var conversions = compilation.Conversions.WithNullability(true);
+                if (RefKind.None != refKind &&
```

Comment:
I'm not sure that's what we want for `out` parameters.
I understand that `out` parameters are just treated as `ref` at the IL level, but at the C# level (which is where the nullable feature operates) we could treat them with variance like return values?

Also, I'd probably expect `in` parameters to behave with variance too.

Reply:
That seems reasonable, although I didn't see it specified anywhere. I'll assume that's the case for now.

Reply:
It would probably be good to record this PR's decision into the design doc.

---
In reply to: [272919639](https://github.com/dotnet/roslyn/pull/34718#discussion_r272919639) [](ancestors = 272919639)